README.md
# MEMO.py

Frontdesk python application to record visits while the person being visited is not available

## Getting Started

These instructions will get you a copy of the project up and running on your local machine for development and testing purposes. See deployment for notes on how to deploy the project on a live system.

### Prerequisites

Python2 but preferably python3,  PyQt5 and its dependecies

## Built With

* [PyQt5](https://www.riverbankcomputing.com/software/pyqt/intro) - The UI framework used

## Authors

* **Bento Lima** - *Front and back end* 

## License

This project is licensed under the MIT License - see the [LICENSE.md](LICENSE.md) file for details

## Acknowledgments

* Saw tons of postits with memos in my coworker's desk, and hated it. Asked management if it would interesting, and they agreed.


history.txt

Recado da recepcao! 

Ola Bento Alves Lima			   

O Sr(a) MEW

Veio Ca
Deseja Falar
e deixou o seguinte recado: 

Mewtwo.

17:27:03
06/11/18
Log Recado da recepcao! 

Ola Bento Alves Lima			   

O Sr(a) Joao

Deseja Falar
deseja ve-lo
e deixou o seguinte recado: 

GUGU.

11:58:51
06/20/18
Log Recado da recepcao! 

Ola Bento Alves Lima			   

O Sr(a) ben

Deseja Falar
e deixou o seguinte recado: 

10.

12:00:46
06/20/18
Log Recado da recepcao! 

Ola Bento Alves Lima			 

O Sr(a) Jose Manuel

Telefonou
Veio Ca
Deseja Falar
deseja ve-lo
Volta a telefonar
Volta Ca
Pede para telefonar
Urgente
e deixou o seguinte recado: 

Ok.

10:25:58
07/04/18
Log Recado da recepcao! 

Ola Bento Alves Lima			 

O Sr(a) Jose Manuel

Telefonou
Veio Ca
Deseja Falar
deseja ve-lo
Volta a telefonar
Volta Ca
Pede para telefonar
Urgente
e deixou o seguinte recado: 

ok.

10:26:37
07/04/18
Log Recado da recepcao! 

Ola Adelaide Lopes Almeida                   

O Sr(a) 


e deixou o seguinte recado: 

ok.

10:27:20
07/04/18
Log Recado da recepcao! 

Ola Bento Alves Lima			 

O Sr(a) AdY

Telefonou
Veio ca
e deixou o seguinte recado: 

Teste.


Log Recado da recepcao! 

Ola Bento Alves Lima			 

O Sr(a) ADY

Telefonou
Veio ca
e deixou o seguinte recado: 

Teste.

09:50:36
07/06/18
Log Recado da recepcao! 

Ola Bento Alves Lima			 

O Sr(a) Teste

Telefonou
Veio ca
Deseja Falar
Deseja Ve-lo
Volta a Telefonar
Volta Ca
Pede para telefonar
Urgente
e deixou o seguinte recado: 

GGGGG.

09:53:10
07/06/18
Log Recado da recepcao! 

Ola Bento Alves Lima			 

O Sr(a) Ola

Telefonou
e deixou o seguinte recado: 

dksfs.

09:53:48
07/06/18
Log Recado da recepcao! 

Ola Bento Alves Lima			 

O Sr(a) Teste3

Urgente
e deixou o seguinte recado: 

dkldsd.

09:54:01
07/06/18
Log Recado da recepcao! 

Ola Bento Alves Lima			 

O Sr(a) hh

Telefonou
Veio ca
e deixou o seguinte recado: 

Teste4.

09:54:39
07/06/18
Log Recado da recepcao! 

Ola Bento Alves Lima			 

O Sr(a) blabla


e deixou o seguinte recado: 

.

09:54:58
07/06/18
Log Recado da recepcao! 

Ola Bento Alves Lima			 

O Sr(a) 


e deixou o seguinte recado: 

dknlkdf.

09:55:09
07/06/18
Log Recado da recepcao! 

Ola Bento Alves Lima			 

O Sr(a) 

Deseja Falar
Pede para telefonar
e deixou o seguinte recado: 

 .

09:55:26
07/06/18

Memo.py
import sys
import smtplib
import time
#from PyQt5 import QtGui,QtCore   ------ Pyqt4 code
from PyQt5.QtWidgets import QMainWindow, QApplication, QPushButton,QLineEdit,QTextEdit,QAction,QCheckBox,QWidget,QComboBox,QMessageBox
from PyQt5.QtCore import pyqtSlot
from PyQt5 import QtGui
from email.mime.text import MIMEText
from email.header import Header



#class Window(QtGui.QMainWindow):
class Window(QMainWindow):

    def __init__(self):
        super(Window,self).__init__()
        self.setGeometry(50,50,300,600)                     #Automatic size=?
        self.setWindowTitle('Recados')
        self.setWindowIcon(QtGui.QIcon('logo.png'))

        #extractAction = QtGui.QAction('&Sair',self)
        #extractAction.setStatusTip('Leave the app!')
        #extractAction.setShortcut('q')
        #extractAction.triggered.connect(self.close_Application)

        #self.statusBar()

        #mainMenu = self.menuBar()
        #fileMenu = mainMenu.addMenu('&File')
        #fileMenu.addAction(extractAction)

        #Import name File
        self.my_dict = {}
        with open('email.txt')as f:
            for line in f:
                ((key,val)) = line.split(',')
                self.my_dict[key] = val
        print (self.my_dict)


        self.home()

        #Populate Combobox
        for k,v in self.my_dict.items():
            self.comboBox1.addItem(k)

#Our Window changing shit
    def home(self):
        #Exit Button
        btn1 = QPushButton('Sair',self)
        btn1.clicked.connect(self.close_Application)
        #btn.resize(100,100)
        btn1.resize(btn1.sizeHint())
        btn1.move(200,500)

        #Send Button
        btn2 = QPushButton('Enviar',self)
        btn2.clicked.connect(self.getTextContent)
        btn2.resize(btn2.sizeHint())
        btn2.move (20,500)

        #Clear Everything button
        btn3 = QPushButton('Limpar tudo',self)
        btn3.clicked.connect(self.clearAll)
        btn3.resize(btn3.sizeHint())
        btn3.move(92,500)

        '''
        #ToolBar
        extractAction = QtGui.QAction(QtGui.QIcon('/Users/bentolima/Desktop/Logo.png'),'Merda de Programa',self)
        extractAction.triggered.connect(self.close_Application)
        self.toolBar = self.addToolBar('extraction')
        self.toolBar.addAction(extractAction)
        '''

        self.textBox2 = QLineEdit(self)
        self.textBox2.setPlaceholderText('Visitante')
        self.textBox2.move (20,210)
        self.textBox2.resize(200,30)

        self.textBox3 = QTextEdit(self)
        self.textBox3.move(20,250)
        self.textBox3.setText('')
        self.textBox3.resize(250,200)


        #checkboxes
        self.checkBox = QCheckBox('Telefonou',self)
        self.checkBox.move(10,10)

        self.checkBox1 = QCheckBox('Veio Ca',self)
        self.checkBox1.move(10,40)

        self.checkBox2 = QCheckBox('Deseja Falar',self)
        self.checkBox2.move(10,70)

        self.checkBox3 = QCheckBox('deseja ve-lo',self)
        self.checkBox3.move(10,100)

        self.checkBox4 = QCheckBox('Volta a telefonar',self)
        self.checkBox4.move(150,10)

        self.checkBox5 = QCheckBox('Volta Ca',self)
        self.checkBox5.move(150,40)

        self.checkBox6 = QCheckBox('Pede para telefonar',self)
        self.checkBox6.move(150,70)

        self.checkBox7 = QCheckBox('Urgente',self)
        self.checkBox7.move(150,100)

        self.comboBox1 = QComboBox(self)
        self.comboBox1.move(20,170)

        self.show()

    def close_Application(self):
        choice = QMessageBox.question(self,'Warning!','Tem a certeza que pretende sair?',
        QMessageBox.Yes|QMessageBox.No)

        if (choice == QMessageBox.Yes):
            sys.exit()
        else:
            pass


    def clearAll(self):                     #bad code over here iteration
        self.textBox2.clear();
        self.textBox3.clear();

        self.checkBox.setChecked(False)
        self.checkBox1.setChecked(False)
        self.checkBox2.setChecked(False)
        self.checkBox3.setChecked(False)
        self.checkBox4.setChecked(False)
        self.checkBox5.setChecked(False)
        self.checkBox6.setChecked(False)
        self.checkBox7.setChecked(False)

    def getTextContent(self):
        #spaguetti code: can't iterate through them
        content = []
        status = []

        content.append(self.textBox2.text())
        content.append(self.textBox3.toPlainText())

        if self.checkBox.isChecked():
            status.append(self.checkBox.text())
        if self.checkBox1.isChecked():
            status.append(self.checkBox1.text())
        if self.checkBox2.isChecked():
            status.append(self.checkBox2.text())
        if self.checkBox3.isChecked():
            status.append(self.checkBox3.text())
        if self.checkBox4.isChecked():
            status.append(self.checkBox4.text())
        if self.checkBox5.isChecked():
            status.append(self.checkBox5.text())
        if self.checkBox5.isChecked():
            status.append(self.checkBox6.text())
        if self.checkBox6.isChecked():
            status.append(self.checkBox7.text())

        #Save content to file
        file = open('testfile.txt','w')
        log = open ('history.txt','a')

        file.write('\n')
        file.write('Recado da recepcao! \n\n')
        file.write('Ola '+ self.comboBox1.currentText()+'\n\nO Sr(a) '+content[0]+'\n\n')
        file.write('\n'.join([str(x) for x in status]))
        file.write('\ne deixou o seguinte recado: \n\n'+content[1]+'.\n\n')
        file.close()
        #Save Log of visits and messages

        log.write('\n')
        log.write('Log Recado da recepcao! \n\n')
        log.write('Ola '+ self.comboBox1.currentText()+'\n\nO Sr(a) '+content[0]+'\n\n')
        log.write('\n'.join([str(x) for x in status]))
        log.write('\ne deixou o seguinte recado: \n\n'+content[1]+'.\n\n')
        log.write(time.strftime('%H:%M:%S\n'))
        log.write(time.strftime('%x'))
        log.close()

        #Send mail
        targetMail = self.my_dict[str(self.comboBox1.currentText())]
        to = targetMail.strip()
        gmail_user = 'secretaria@bento.cv'
        gmail_pwd = 'bento123.'
        smtpserver = smtplib.SMTP("mail.bento.cv",587)
        smtpserver.ehlo()
        smtpserver.starttls()
        smtpserver.ehlo()
        smtpserver.login(gmail_user, gmail_pwd)
        header = 'To:' + to + '\n' + 'From: ' + gmail_user + '\n' + 'Subject: Recado Da Recepcao\n'
        print (header)
        with open('testfile.txt','rb') as fp:
            msg = MIMEText(fp.read())
        smtpserver.sendmail(str(gmail_user), str(to),header+str(msg))
        print ('done!')
        smtpserver.close()
        self.clearAll()






def main():
    #app = QtGui.QApplication(sys.argv)
    app = QApplication(sys.argv)
    Gui = Window()
    sys.exit(app.exec_())

if __name__ == '__main__':
    main()


send.py
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
from email.mime.base import MIMEBase
from email import encoders

email_user = 'secretaria@bento.cv'
email_password = 'bento123.'
email_send = 'bento.lima@bento.cv'

subject = 'teste'

msg = MIMEMultipart()
msg['From'] = email_user
msg['To'] = email_send
msg['Subject'] = subject

body = 'Hi there, sending this email from Python!'
msg.attach(MIMEText(body,'plain'))

#filename='filename'
#attachment  =open(filename,'rb')

#part = MIMEBase('application','octet-stream')
#part.set_payload((attachment).read())
#encoders.encode_base64(part)
#part.add_header('Content-Disposition',"attachment; filename= "+filename)

#msg.attach(part)
#text = msg.as_string()
server = smtplib.SMTP('smtp.bento.com',587)
server.starttls()
server.login(email_user,email_password)


server.sendmail(email_user,email_send,text)
server.quit()

design.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'design.ui'
#
# Created by: PyQt5 UI code generator 5.10.1
#
# WARNING! All changes made in this file will be lost!
import sys
import smtplib
import time
from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtWidgets import QMainWindow, QApplication, QPushButton,QLineEdit,QTextEdit,QAction,QCheckBox,QWidget,QComboBox,QMessageBox
from PyQt5.QtCore import pyqtSlot
from PyQt5 import QtGui
from email.mime.text import MIMEText
from email.header import Header

class Ui_Form(QMainWindow):

    def __init__(self):
        super(Ui_Form,self).__init__()
        #self.setWindowIcon(QtGui.QIcon('Logo.png'))
        self.my_dict = {}
        with open('email.txt')as f:
            for line in f:
                ((key,val)) = line.split(',')
                self.my_dict[key] = val
        #print (self.my_dict)


        self.setupUi(Form)

        #Populate Combobox
        for k,v in self.my_dict.items():
            self.comboBox1.addItem(k)


    def setupUi(self,Form):

        Form.setObjectName("Form")
        Form.resize(386, 413)
        self.gridLayout = QtWidgets.QGridLayout(Form)
        self.gridLayout.setObjectName("gridLayout")
        self.verticalLayout = QtWidgets.QVBoxLayout()
        self.verticalLayout.setObjectName("verticalLayout")


        self.comboBox1 = QtWidgets.QComboBox(Form)
        self.comboBox1.setObjectName("comboBox1")
        self.verticalLayout.addWidget(self.comboBox1)


        self.textBox2 = QtWidgets.QLineEdit(Form)
        self.textBox2.setObjectName("textBox2")
        self.textBox2.setPlaceholderText('Visitante')


        self.verticalLayout.addWidget(self.textBox2)
        self.gridLayout.addLayout(self.verticalLayout, 0, 0, 1, 4)
        self.verticalLayout_2 = QtWidgets.QVBoxLayout()
        self.verticalLayout_2.setObjectName("verticalLayout_2")


        self.checkBox_0 = QtWidgets.QCheckBox(Form)
        self.checkBox_0.setObjectName("checkBox_0")
        self.verticalLayout_2.addWidget(self.checkBox_0)


        self.checkBox_1 = QtWidgets.QCheckBox(Form)
        self.checkBox_1.setObjectName("checkBox_1")
        self.verticalLayout_2.addWidget(self.checkBox_1)


        self.checkBox_2 = QtWidgets.QCheckBox(Form)
        self.checkBox_2.setObjectName("checkBox_2")
        self.verticalLayout_2.addWidget(self.checkBox_2)


        self.checkBox_3 = QtWidgets.QCheckBox(Form)
        self.checkBox_3.setObjectName("checkBox_3")
        self.verticalLayout_2.addWidget(self.checkBox_3)


        self.gridLayout.addLayout(self.verticalLayout_2, 1, 0, 1, 2)
        self.verticalLayout_4 = QtWidgets.QVBoxLayout()
        self.verticalLayout_4.setObjectName("verticalLayout_4")


        self.checkBox_4 = QtWidgets.QCheckBox(Form)
        self.checkBox_4.setObjectName("checkBox_4")
        self.verticalLayout_4.addWidget(self.checkBox_4)


        self.checkBox_5 = QtWidgets.QCheckBox(Form)
        self.checkBox_5.setObjectName("checkBox_5")
        self.verticalLayout_4.addWidget(self.checkBox_5)


        self.checkBox_6 = QtWidgets.QCheckBox(Form)
        self.checkBox_6.setObjectName("checkBox_6")
        self.verticalLayout_4.addWidget(self.checkBox_6)


        self.checkBox_7 = QtWidgets.QCheckBox(Form)
        self.checkBox_7.setObjectName("checkBox_7")
        self.verticalLayout_4.addWidget(self.checkBox_7)


        self.gridLayout.addLayout(self.verticalLayout_4, 1, 2, 1, 2)
        self.horizontalLayout = QtWidgets.QHBoxLayout()
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.textBox3 = QtWidgets.QTextEdit(Form)
        self.textBox3.setObjectName("textBox3")
        self.horizontalLayout.addWidget(self.textBox3)
        self.gridLayout.addLayout(self.horizontalLayout, 2, 0, 1, 4)
        self.verticalLayout_3 = QtWidgets.QVBoxLayout()
        self.verticalLayout_3.setObjectName("verticalLayout_3")
        self.btn2 = QtWidgets.QPushButton(Form)
        self.btn2.setObjectName("btn2")
        self.verticalLayout_3.addWidget(self.btn2)
        self.gridLayout.addLayout(self.verticalLayout_3, 3, 0, 1, 1)
        self.verticalLayout_5 = QtWidgets.QVBoxLayout()
        self.verticalLayout_5.setObjectName("verticalLayout_5")
        self.btn3 = QtWidgets.QPushButton(Form)
        self.btn3.setObjectName("btn3")
        self.verticalLayout_5.addWidget(self.btn3)
        self.gridLayout.addLayout(self.verticalLayout_5, 3, 1, 1, 2)
        self.verticalLayout_6 = QtWidgets.QVBoxLayout()
        self.verticalLayout_6.setObjectName("verticalLayout_6")
        self.btn1 = QtWidgets.QPushButton(Form)
        self.btn1.setObjectName("btn1")
        self.verticalLayout_6.addWidget(self.btn1)
        self.gridLayout.addLayout(self.verticalLayout_6, 3, 3, 1, 1)

        self.retranslateUi(Form)
        QtCore.QMetaObject.connectSlotsByName(Form)

        #self.show()
        #My code Goes here

    def retranslateUi(self, Form):
        _translate = QtCore.QCoreApplication.translate
        Form.setWindowIcon(QtGui.QIcon('logo.png'))
        Form.setWindowTitle(_translate("Form", "Recados"))
        self.checkBox_0.setText(_translate("Form", "Telefonou"))
        self.checkBox_1.setText(_translate("Form", "Veio ca"))
        self.checkBox_2.setText(_translate("Form", "Deseja Falar"))
        self.checkBox_3.setText(_translate("Form", "Deseja Ve-lo"))
        self.checkBox_4.setText(_translate("Form", "Volta a Telefonar"))
        self.checkBox_5.setText(_translate("Form", "Volta Ca"))
        self.checkBox_6.setText(_translate("Form", "Pede para telefonar"))
        self.checkBox_7.setText(_translate("Form", "Urgente"))
        self.btn2.setText(_translate("Form", "Enviar"))
        self.btn2.clicked.connect(self.getTextContent)
        self.btn3.setText(_translate("Form", "Limpar tudo"))
        self.btn3.clicked.connect(self.clearAll)
        self.btn1.setText(_translate("Form", "Sair"))
        self.btn1.clicked.connect(self.close_Application)

    def clearAll(self):                     #bad code over here iteration
        self.textBox2.clear();
        self.textBox3.clear();

        self.checkBox_0.setChecked(False)
        self.checkBox_1.setChecked(False)
        self.checkBox_2.setChecked(False)
        self.checkBox_3.setChecked(False)
        self.checkBox_4.setChecked(False)
        self.checkBox_5.setChecked(False)
        self.checkBox_6.setChecked(False)
        self.checkBox_7.setChecked(False)

    def close_Application(self):
        choice = QMessageBox.question(self,'Warning!',"Tem a certeza que pretende sair?", QMessageBox.Yes|QMessageBox.No,QMessageBox.No)

        if (choice == QMessageBox.Yes):
            sys.exit()
        else:
            pass

    def getTextContent(self):
        #spaguetti code: can't iterate through them
        content = []
        status = []

        content.append(self.textBox2.text())
        content.append(self.textBox3.toPlainText())

        if self.checkBox_0.isChecked():
            status.append(self.checkBox_0.text())
        if self.checkBox_1.isChecked():
            status.append(self.checkBox_1.text())
        if self.checkBox_2.isChecked():
            status.append(self.checkBox_2.text())
        if self.checkBox_3.isChecked():
            status.append(self.checkBox_3.text())
        if self.checkBox_4.isChecked():
            status.append(self.checkBox_4.text())
        if self.checkBox_5.isChecked():
            status.append(self.checkBox_5.text())
        if self.checkBox_6.isChecked():
            status.append(self.checkBox_6.text())
        if self.checkBox_7.isChecked():
            status.append(self.checkBox_7.text())

        #Save content to file
        file = open('testfile.txt','w')
        log = open ('history.txt','a')

        file.write('\n')
        file.write('Recado da recepcao! \n\n')
        file.write('Ola '+ self.comboBox1.currentText()+'\n\nO Sr(a) '+content[0]+'\n\n')
        file.write('\n'.join([str(x) for x in status]))
        file.write('\ne deixou o seguinte recado: \n\n'+content[1]+'.\n\n')
        file.close()
        #Save Log of visits and messages

        log.write('\n')
        log.write('Log Recado da recepcao! \n\n')
        log.write('Ola '+ self.comboBox1.currentText()+'\n\nO Sr(a) '+content[0]+'\n\n')
        log.write('\n'.join([str(x) for x in status]))
        log.write('\ne deixou o seguinte recado: \n\n'+content[1]+'.\n\n')
        log.write(time.strftime('%H:%M:%S\n'))
        log.write(time.strftime('%x'))
        log.close()

        #Send mail
        targetMail = self.my_dict[str(self.comboBox1.currentText())]
        to = targetMail.strip()
        gmail_user = 'secretaria@bento.cv'
        gmail_pwd = 'bento123.'
        smtpserver = smtplib.SMTP("mail.bento.cv",587)
        smtpserver.ehlo()
        smtpserver.starttls()
        smtpserver.ehlo()
        smtpserver.login(gmail_user, gmail_pwd)
        header = 'To:' + to + '\n' + 'From: ' + gmail_user + '\n' + 'Subject: Recado Da Recepcao\n'
        print (header)
        with open('testfile.txt','rb') as fp:
            msg = MIMEText(fp.read())
        smtpserver.sendmail(str(gmail_user), str(to),header+str(msg))
        print ('done!')
        smtpserver.close()
        self.clearAll()

if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    Form = QtWidgets.QWidget()
    ui = Ui_Form()
    #ui.setupUi(Form)
    Form.show()
    sys.exit(app.exec_())


pyqt_tut.py
import sys
import smtplib
from PyQt4 import QtGui,QtCore
from email.mime.text import MIMEText
from email.header import Header



class Window(QtGui.QMainWindow):

    def __init__(self):
        super(Window,self).__init__()
        self.setGeometry(50,50,300,600)                     #Automatic size=?
        self.setWindowTitle('PyQt Tutorial')
        self.setWindowIcon(QtGui.QIcon('/Users/bentolima/Desktop/Logo.png'))

        extractAction = QtGui.QAction('&Exit',self)
        extractAction.setStatusTip('Leave the app!')
        extractAction.setShortcut('q')
        extractAction.triggered.connect(self.close_Application)

        self.statusBar()

        mainMenu = self.menuBar()
        fileMenu = mainMenu.addMenu('&File')
        fileMenu.addAction(extractAction)

        #Import name File
        dict = {}
        with open('nomes.txt')as f:
            for line in f:
                (key,val) = line.split(',')
                dict[key] = val
        print (dict)


        self.home()

        #Populate Combobox
        for k,v in dict.items():
            self.comboBox1.addItem(k)

#Our Window changing shit
    def home(self):
        #Exit Button
        btn1 = QtGui.QPushButton('Exit',self)
        btn1.clicked.connect(self.close_Application)
        #btn.resize(100,100)
        btn1.resize(btn1.sizeHint())
        btn1.move(200,500)

        #Send Button
        btn2 = QtGui.QPushButton('Enviar',self)
        btn2.clicked.connect(self.getTextContent)
        #btn2.clicked.connect(self.getBoxStatus)
        btn2.resize(btn2.sizeHint())
        btn2.move (20,500)

        #Clear Everything button
        btn3 = QtGui.QPushButton('Limpar tudo',self)
        btn3.clicked.connect(self.clearAll)
        btn3.resize(btn3.sizeHint())
        btn3.move(92,500)

        '''
        #ToolBar
        extractAction = QtGui.QAction(QtGui.QIcon('/Users/bentolima/Desktop/Logo.png'),'Merda de Programa',self)
        extractAction.triggered.connect(self.close_Application)
        self.toolBar = self.addToolBar('extraction')
        self.toolBar.addAction(extractAction)



        self.textBox1 =  QtGui.QTextEdit(self)
        self.textBox1.setText('Visitado')
        self.textBox1.move(20,170)
        self.textBox1.resize(200,30)
        '''

        self.textBox2 = QtGui.QTextEdit(self)
        self.textBox2.setText('Visitante')
        self.textBox2.move (20,210)
        self.textBox2.resize(200,30)

        self.textBox3 = QtGui.QTextEdit(self)
        self.textBox3.move(20,250)
        self.textBox3.setText('Mensagem')
        self.textBox3.resize(250,200)


        #checkboxes
        self.checkBox = QtGui.QCheckBox('Telefonou',self)
        self.checkBox.move(10,10)
        #self.checkBox.stateChanged.connect(self.printsomeshit) ##dunno yet

        self.checkBox1 = QtGui.QCheckBox('Veio Ca',self)
        self.checkBox1.move(10,40)
        #self.checkBox1.stateChanged.connect(self.printsomeshit) ##dunno yet

        self.checkBox2 = QtGui.QCheckBox('Deseja Falar',self)
        self.checkBox2.move(10,70)
        #self.checkBox2.stateChanged.connect(self.printsomeshit) ##dunno yet

        self.checkBox3 = QtGui.QCheckBox('deseja ve-lo',self)
        self.checkBox3.move(10,100)
        #self.checkBox3.stateChanged.connect(self.printsomeshit) ##dunno yet

        self.checkBox4 = QtGui.QCheckBox('Volta a telefonar',self)
        self.checkBox4.move(150,10)
        #self.checkBox4.stateChanged.connect(self.printsomeshit) ##dunno yet

        self.checkBox5 = QtGui.QCheckBox('Volta Ca',self)
        self.checkBox5.move(150,40)
        #self.checkBox5.stateChanged.connect(self.printsomeshit) ##dunno yet

        self.checkBox6 = QtGui.QCheckBox('Pede para telefonar',self)
        self.checkBox6.move(150,70)
        #self.checkBox6.stateChanged.connect(self.printsomeshit) ##dunno yet

        self.checkBox7 = QtGui.QCheckBox('Urgente',self)
        self.checkBox7.move(150,100)
        #self.checkBox7.stateChanged.connect(self.printsomeshit) ##dunno yet

        self.comboBox1 = QtGui.QComboBox(self)
        self.comboBox1.move(20,170)

        self.show()

    def close_Application(self):
        #print('Good Bye!')
        #sys.exit()
        choice = QtGui.QMessageBox.question(self,'Warning!','Are you sure you want to leave?',
        QtGui.QMessageBox.Yes|QtGui.QMessageBox.No)

        if (choice == QtGui.QMessageBox.Yes):
            sys.exit()
        else:
            pass


    def clearAll(self):                     #bad code over here iteration
        #self.textBox1.clear();
        self.textBox2.clear();
        self.textBox3.clear();

        self.checkBox.setChecked(False)
        self.checkBox1.setChecked(False)
        self.checkBox2.setChecked(False)
        self.checkBox3.setChecked(False)
        self.checkBox4.setChecked(False)
        self.checkBox5.setChecked(False)
        self.checkBox6.setChecked(False)
        self.checkBox7.setChecked(False)

        #iterate through Qcheckboxes/QtextEdit identify the selected ones/messages
        #print selected and content to terminal

    def getTextContent(self):
        #bad code here cant iterate through them
        content = []
        status = []

        content.append(self.textBox2.toPlainText())
        content.append(self.textBox3.toPlainText())

        if self.checkBox.isChecked():
            status.append(self.checkBox.text())
        if self.checkBox1.isChecked():
            status.append(self.checkBox1.text())
        if self.checkBox2.isChecked():
            status.append(self.checkBox2.text())
        if self.checkBox3.isChecked():
            status.append(self.checkBox3.text())
        if self.checkBox4.isChecked():
            status.append(self.checkBox4.text())
        if self.checkBox5.isChecked():
            status.append(self.checkBox.text())
        if self.checkBox5.isChecked():
            status.append(self.checkBox.text())
        if self.checkBox6.isChecked():
            status.append(self.checkBox6.text())

        #Store the message in a string and return it.
        '''
        print ('\n\n')
        print ('Recado da recepcao! \n\n')
        print('Ola '+self.comboBox1.currentText()+ '\n\nO Sr(a) '+content[0]+'\n\n')

        print '\n'.join([str(x) for x in status])

        print('\ne deixou o seguinte recado: \n\n'+content[1]+'.\n\n')
        '''

        #Save content to file
        file = open('testfile.txt','w')

        file.write('\n')
        file.write('Recado da recepcao! \n\n')
        file.write('Ola '+ self.comboBox1.currentText()+'\n\nO Sr(a) '+content[0]+'\n\n')
        file.write('\n'.join([str(x) for x in status]))
        file.write('\ne deixou o seguinte recado: \n\n'+content[1]+'.\n\n')
        file.close()

        #Send mail

        #to = self.comboBox1.currentText()
        to = 'bento.lima@bento.cv'
        gmail_user = 'secretaria@bento.cv'
        gmail_pwd = 'bento123.'
        smtpserver = smtplib.SMTP("mail.bento.cv",587)
        smtpserver.ehlo()
        smtpserver.starttls()
        smtpserver.ehlo()
        smtpserver.login(gmail_user, gmail_pwd)
        header = 'To:' + to + '\n' + 'From: ' + gmail_user + '\n' + 'Subject: Recado Da Recepcao\n'
        print header
        with open('testfile.txt','rb') as fp:
            msg = MIMEText(fp.read())
        smtpserver.sendmail(str(gmail_user), str(to),header+str(msg))
        print 'done!'
        smtpserver.close()






def main():
    app = QtGui.QApplication(sys.argv)
    Gui = Window()
    sys.exit(app.exec_())

if __name__ == '__main__':
    main()
